### DISCUSSION TITLE
Where exists on relations?

### DISCUSSION THREAD

**Discussion by gufoe** [SeaQL/sea-orm#1106] [Q&A] [answered]
Hello, I am unable to find any documentation on how to filter a query based on the existance of its relations.

In Laravel where I come from, I can do things like:
```php
User::whereHas('sessions', function($ingredient_subquery) {
  $session->where('token', '=', 'xxx');
})->first()
```

Which gives
```sql
(select * from users where exists(select 1 from sessions where user_id = users.id))
```

I managed to achieve something similar:
```rust
let sessions_query = entity::sessions::Entity::find()
        .select_only()
        .column(entity::sessions::Column::UserId)
        .filter(entity::sessions::Column::Token.eq("xxx"));

let users = entity::users::Entity::find()
  .filter(entity::users::Column::Id.in_subquery(sessions_query.as_query().clone()));
```

but it uses the where in clause instead of the exists which is way faster on big tables.
```sql
(select * from users where id in (select user_id from sessions where user_id = users.id))
```

I believe this is possible but I searched in examples, issues and discussions and could not find an example for this.

**Comment by billy1624** [comment, marked as answer]
Hey @gufoe, welcome!! Thanks for asking. It has been added to SeaQuery recently and will be available in the next release.
- https://github.com/SeaQL/sea-query/pull/379

**Reply by gufoe** [reply]
Awesome thank you!
